qemu/tests/balloon_check.py

```diff
@@ -87,6 +88,54 @@ def memory_check(self, step, ballooned_mem):
                 raise exceptions.TestFail("Balloon test failed %s" % step)
         return (mmem, gmem)
 
+    def memory_status_check(self, keyname):
+        """
+        Check memory status via balloon service.
+
+        :param keyname: key name of the output of the 'qom-get' property.
+        :return: if memory stastics can be got via balloon service, reture True,
+                 else return False.
+        :rtype: bool
+        """
+        check_mem_ratio = float(self.params.get("check_mem_ratio", 0.1))
+        sleep_time = int(self.params.get("polling_sleep_time", 20))
+        base_path = self.params.get("base_path", "/machine/peripheral/")
+        device = self.params.get("balloon", "balloon0")
+        device_path = os.path.join(base_path, device)
+        polling_interval = int(self.params.get("polling_interval", 2))
+        set_balloon_property = self.params.get("set_balloon_property",
+                                               "guest-stats-polling-interval")
+        get_balloon_property = self.params.get("get_balloon_property", "guest-stats")
+
+        error_context.context("Get memory from guest", logging.info)
+        if keyname == "stat-free-memory":
+            guest_mem = self.get_guest_free_mem(self.vm)
+        elif keyname == "stat-total-memory":
+            guest_mem = self.get_vm_mem(self.vm)
+
+        error_context.context("Get memory from qmp", logging.info)
+        self.vm.monitor.qom_set(device_path, set_balloon_property, polling_interval)
+        time.sleep(sleep_time)
+        get_polling_output = self.vm.monitor.qom_get(device_path,
+                                                     get_balloon_property)
+        stat_memory_qmp = get_polling_output['stats'][keyname]
+
+        if stat_memory_qmp == 18446744073709551615:
+            memory_status_enable = False
+            logging.info("Memory satus is not enabled")
+        else:
+            memory_status_enable = True
+            error_context.context("Compare memory from guest with qmp", logging.info)
+            stat_memory_qmp = "%sB" % stat_memory_qmp
+            stat_memory_qmp = int(float(utils_misc.normalize_data_size(
+                                       stat_memory_qmp, order_magnitude="M")))
+            if (abs(guest_mem - stat_memory_qmp) / guest_mem) > check_mem_ratio:
```

Comment:
guest_mem and stat_memory_qmp both are int type, '/' is a Integer option, the "abs(guest_mem - stat_memory_qmp) / guest_mem" is a int type, but check_mem_ratio is  float type, change the abs(guest_mem - stat_memory_qmp) to float type is necessary. 
Like if float(abs(guest_mem - stat_memory_qmp)) / guest_mem > check_mem_ratio: